modules/social_features/social_event/modules/social_event_an_enroll/src/Form/EventAnEnrollActionForm.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace Drupal\social_event_an_enroll\Form;
+
+use Drupal\social_event\Form\EnrollActionForm;
+use Drupal\Core\Cache\Cache;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Url;
+use Drupal\node\Entity\Node;
+
+/**
+ * Class EventAnEnrollActionForm.
+ *
+ * @package Drupal\social_event_an_enroll\Form
+ */
+class EventAnEnrollActionForm extends EnrollActionForm {
```

Comment:
Why are we extending the EnrollActionForm class? I understand that this is the enrollment but for anonymous users. However, we don't reuse any logic from the EnrollActionForm class which suggests that these are two entirely separate forms and shouldn't inherit from eachother.

Reply:
This extending is actually needed and useful because we reuse method `eventHasBeenFinished()` and also dependency injections from parent class like `$this->entityStorage`, `$this->currentUser` etc